src/core/folder.h

```diff
@@ -179,6 +183,8 @@ private Q_SLOTS:
     bool defer_content_test : 1;
 
     static std::unordered_map<FilePath, std::weak_ptr<Folder>, FilePathHash> cache_;
+    static FilePath cutFilesDirPath_;
+    static std::shared_ptr<const HashSet> cutFilesHashSet_;
```

Comment:
Actually, I'm also suspicious about the static-ness of `cache_`, which may explain https://github.com/lxde/pcmanfm-qt/issues/508. I should look into that... but it isn't related to this PR.